### PR TITLE
Switch between prefetching (as implemented now), and no prefetching at all

### PR DESCRIPTION
**Pull request by ArturAkh**
Related to https://github.com/HEPCompSim/DistCacheSim/pull/34

A switch back to a boolean choice between the two possibilities.

### CODE REVIEW

**Review by HerrHorizontal — CHANGES_REQUESTED** [review]
This looks good! But I think we should change the default behaviour.

File: src/SimpleSimulator.cpp [hunk 1]
```diff
@@ -28,6 +28,7 @@ namespace po = boost::program_options;
 std::map<std::shared_ptr<wrench::StorageService>, LRU_FileList> SimpleSimulator::global_file_map;
 std::mt19937 SimpleSimulator::gen(42);  // random number generator
 bool SimpleSimulator::use_blockstreaming = true;   // flag to chose between simulated job types: streaming or copy jobs
+bool SimpleSimulator::prefetching_on = false;   // flag to enable prefetching during streaming
```
[the remark this answers is not in the document]
Comment (ArturAkh) [reply src/SimpleSimulator.cpp]: Sure, done: 0b44488d11d86631af9e8ff4adc1ed9eca223610

File: src/SimpleSimulator.cpp [hunk 2]
```diff
@@ -70,6 +71,7 @@ po::variables_map process_program_options(int argc, char** argv) {
     size_t duplications = 1;
 
     bool no_blockstreaming = false;
+    bool prefetch_on = false;
```
[the remark this answers is not in the document]
Comment (ArturAkh) [reply src/SimpleSimulator.cpp]: Done: 0b44488d11d86631af9e8ff4adc1ed9eca223610

[the remark this answers is not in the document]
Comment (ArturAkh) [reply src/SimpleSimulator.cpp]: ok have to rework it a bit more

**Comment by ArturAkh** [comment]
Changed the behaviour of the prefetching switch: a7422b22c0bbfc1e455a6ec778c3c30fc731da85

Also tested it and it works as expected.

File: src/SimpleSimulator.cpp [hunk 3]
```diff
@@ -281,6 +284,10 @@ int main(int argc, char **argv) {
     // Flags to turn on/off blockwise streaming of input-files
     SimpleSimulator::use_blockstreaming = !(vm["no-streaming"].as<bool>());
 
+    // Flags to turn refetching for streaming of input-files
+    std::cerr << "Prefetching switch off?: " << vm["prefetch-off"].as<bool>() << std::endl;
```
Comment (HerrHorizontal) [inline src/SimpleSimulator.cpp:288]: You want to keep this print-out?

**Review by HerrHorizontal — COMMENTED** [review]
So it looks good! But do you want to keep this print-out, or did you forget to remove it after debugging?

File: src/SimpleSimulator.cpp (hunk 3 above)
Comment (ArturAkh) [reply src/SimpleSimulator.cpp:288]: I'd rather like to keep it ^^ But if you would like to remove it, that's fine with mee, too.

**Pull request merged**